### PR TITLE
chore(containers): expose state and actions APIs

### PR DESCRIPTION
**Pull request by jmfrancois**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

At the moment to get access to the API you must import with the complete path:

```
import { getStateAccessors, getStateProps } from 'react-talend-containers/lib/state';
const propsA = getStateAccessors(dispatch, info.name, info.id, DEFAULT_STATE);
const propsB = getStateProps(state, info.name, info.id);
```

**What is the new behavior?**

You can now import it directly through react-talend-containers

```javascript
import { componentState } from 'react-talend-containers';
const props = componentState.getAccessors(dispatch, info.name, info.id, DEFAULT_STATE);
const props = componentState.getProps(state, info.name, info.id);
```

Same for actions API.

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://147.talend.surge.sh)

File: packages/cmf/src/reducers/componentsReducers.js [hunk 1]
```diff
@@ -14,17 +15,31 @@ export const defaultState = new Map();
...
 		}
+
 		return state.setIn(
 			[action.componentName, action.key],
 			new Map()
 		);
 	case ACTIONS.componentsActions.COMPONENT_MERGE_STATE:
+		if (!state.getIn([action.componentName, action.key])) {
+			let msg = 'The component state can\'t be merged since the ';
+			msg += `${action.componentName}, ${action.key} association doesn't exist.`;
```
Comment (talend-glorieux) [inline packages/cmf/src/reducers/componentsReducers.js]: Template string for the entire string sound's better than concatenation. Less assignation ;)

File: packages/containers/src/actionAPI.js [hunk 2]
```diff
@@ -1,39 +1,38 @@
...
+
+	let tmpIds = ids;
+	if (typeof ids === 'string') {
+		tmpIds = [ids];
+	}
+
+	const infos = tmpIds.map(id => api.action.getActionInfo(context, id));
+	infos.forEach((info) => {
+		info.model = model; // eslint-disable-line no-param-reassign
+		info.onClick = (event, data) => { // eslint-disable-line no-param-reassign
```
Comment (talend-glorieux) [inline packages/containers/src/actionAPI.js]: I don't like seeing `eslint-disable-line` comments

**Comment by acateland** [comment]
I think we should go further, `react-talend-containers` use react-cmf wich in turns declare store instance.

```javascript
import { componentState } from 'react-talend-containers';
const props = componentState.getAccessors(info.name, info.id, DEFAULT_STATE);
const props = componentState.getProps(info.name, info.id);
```

**Comment by jmfrancois** [comment]
Yes but at the moment all the functions are pure, so creating this kind of underhood links make it hard to tests.
Kwowing this API is used in mapStateToProps and mapDispatchToProps, I really don't want that at the moment.

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://147.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://147.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://147.talend.surge.sh)

**Pull request merged**